pkg/controller/statusmanager/status_manager.go

```diff
@@ -107,55 +112,130 @@ func (status *StatusManager) deleteRelatedObjectsNotRendered(co *configv1.Cluste
 	}
 }
 
-// Set updates the ClusterOperator.Status with the provided conditions
-func (status *StatusManager) set(reachedAvailableLevel bool, conditions ...configv1.ClusterOperatorStatusCondition) {
+// Set updates the operator and clusteroperator statuses with the provided conditions.
+func (status *StatusManager) set(reachedAvailableLevel bool, conditions ...operv1.OperatorCondition) {
+	var operStatus *operv1.NetworkStatus
+
+	// Set status on the network.operator object
 	err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
-		co := &configv1.ClusterOperator{ObjectMeta: metav1.ObjectMeta{Name: status.name}}
-		err := status.client.Get(context.TODO(), types.NamespacedName{Name: status.name}, co)
-		isNotFound := errors.IsNotFound(err)
-		if err != nil && !isNotFound {
+		oc := &operv1.Network{ObjectMeta: metav1.ObjectMeta{Name: names.OPERATOR_CONFIG}}
+		err := status.client.Get(context.TODO(), types.NamespacedName{Name: names.OPERATOR_CONFIG}, oc)
+		if err != nil && !errors.IsNotFound(err) {
+			// Should never happen outside of unit tests
 			return err
 		}
 
-		oldStatus := co.Status.DeepCopy()
-		status.deleteRelatedObjectsNotRendered(co)
-		if status.relatedObjects != nil {
-			co.Status.RelatedObjects = status.relatedObjects
-		}
+		oldStatus := oc.Status.DeepCopy()
 
-		if reachedAvailableLevel {
-			if releaseVersion := os.Getenv("RELEASE_VERSION"); len(releaseVersion) > 0 {
-				co.Status.Versions = []configv1.OperandVersion{
-					{Name: "operator", Version: releaseVersion},
-				}
-			} else {
-				co.Status.Versions = nil
+		oc.Status.Version = os.Getenv("RELEASE_VERSION")
+
+		// If we have no conditions at all, pull them from the ClusterOperator object.
```

Comment:
why?

Reply:
because we're upgrading from an existing cluster. updated the comment.